assets/javascripts/lib/apply-reply.js

```diff
@@ -41,13 +38,20 @@ export default function (replyId, replyTitle, replyContent, model) {
     }
   }
 
-  // Finally insert canned reply.
-  model.appEvents.trigger("composer:insert-block", replyContent);
+  return { replyTitle, replyContent };
+}
+
+export function insertReplyIntoComposer({ replyTitle, replyContent }) {
```

Comment:
Not sure how I feel about a function that has to be called via `call`/`apply` in a specific context. 🤔 

Imo it would be easier to follow if it was inlined instead (it's used in two places only if I saw correctly)
Alternatively it could have an explicit `context` argument, which would replace `this`.

Reply:
I've added an explicit `context` argument to make thing more clear. My original goal when I extracted this function was to prepare the code to allow the use of canned replies in other places besides the composer (specifically the Edit Staff Notice Modal).

I haven't added the functionality yet because there is no connector there and I don't want to override the template.

Changed in https://github.com/discourse/discourse-canned-replies/pull/99/commits/4c48b9685b9ea9851599bf3392084b319b9165da